File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& arr, int target) {
        int l = 0 , h = arr.size()-1;
        while(l < h){
            int sum = arr[l] + arr[h];
            if(sum > target)
                h--;
            else if(sum < target)
                l++;
            else 
                return {l+1 , h+1};
        }
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-1.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> mp = new HashMap<>();
        for(String str : strs){
            char []arr = str.toCharArray();
            Arrays.sort(arr);
            String s = new String(arr);

            mp.putIfAbsent(s,new ArrayList<>());
            mp.get(s).add(str);
        }
        return new ArrayList<>(mp.values());
    }
}
